### PR TITLE
*: limit the error message and log length; fix query cost check

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

1. limit the error message and log length for DB operations
2. fix query cost check

### What is changed and how it works?

1. truncate queries and arguments for log and error message
2. fix `if cost > 1` check

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@amyangfei @WangXiangUSTC PTAL

File: pkg/baseconn/conn.go [hunk 1]
```diff
@@ -89,13 +94,18 @@ func (conn *BaseConn) QuerySQL(tctx *tcontext.Context, query string, args ...int
...
 	rows, err := conn.DB.QueryContext(tctx.Context(), query, args...)
 
 	if err != nil {
-		tctx.L().Error("query statement failed", zap.String("query", query), zap.Reflect("argument", args), log.ShortError(err))
-		return nil, terror.ErrDBQueryFailed.Delegate(err, query)
+		tctx.L().Error("query statement failed",
+			zap.String("query", utils.TruncateString(query, stringLenLimit)),
+			zap.Reflect("argument", utils.TruncateInterface(args, stringLenLimit)),
+			log.ShortError(err))
+		return nil, terror.ErrDBQueryFailed.Delegate(err, utils.TruncateString(query, stringLenLimit))
```
Comment (WangXiangUSTC) [inline pkg/baseconn/conn.go]: how about remain the whole error message, only do TruncateString when need print it

File: loader/db.go [hunk 2]
```diff
@@ -67,14 +75,20 @@ func (conn *Conn) querySQL(ctx *tcontext.Context, query string, args ...interfac
 			if err == nil {
 				cost := time.Since(startTime)
 				queryHistogram.WithLabelValues(conn.cfg.Name).Observe(cost.Seconds())
-				if cost > 1 {
-					ctx.L().Warn("query statement", zap.String("query", query), zap.Reflect("argument", args), zap.Duration("cost time", cost))
+				if cost.Seconds() > 1 {
+					ctx.L().Warn("query statement",
+						zap.String("query", utils.TruncateString(query, stringLenLimit)),
+						zap.Reflect("argument", utils.TruncateInterface(args, stringLenLimit)),
```
Comment (WangXiangUSTC) [inline loader/db.go]: seems we can use zap.String

File: pkg/utils/common.go [hunk 3]
```diff
@@ -33,8 +33,9 @@ import (
 // move to tidb-tools later
 
 const (
-	maxRetryCount = 3
-	retryTimeout  = 3 * time.Second
+	maxRetryCount  = 3
+	retryTimeout   = 3 * time.Second
+	stringLenLimit = 1024
```
Comment (WangXiangUSTC) [inline pkg/utils/common.go]: can we only define it in untils/string.go, and use `-1` as argument in `TruncateString` if use default stringLenLimit?

File: pkg/baseconn/conn.go (hunk 1 above)
Comment (csuzhangxc) [reply pkg/baseconn/conn.go]: the error message can be logged/sent by RPC in many places, it's hard to find out all of them, and ensure truncate it in later usage.

File: pkg/utils/string_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		{v: 123, n: 0, expect: "..."},
+		{v: 123, n: 3, expect: "123"},
+		{v: compose{b: true, i: 123, s: "abc"}, n: 3, expect: "{b:..."},
+		{v: compose{b: true, i: 123, s: "abc"}, n: 9, expect: "{b:true i..."},
+		{v: compose{b: true, i: 123, s: "abc"}, n: 99, expect: "{b:true i:123 s:abc}"},
+		{v: []string{"abc", "123"}, n: 3, expect: "[ab..."},
+		{v: []string{"abc", "123"}, n: 9, expect: "[abc 123]"},
+		{v: []compose{{b: true, i: 123, s: "abc"}, {b: false, i: 456, s: "def"}}, n: 3, expect: "[{b..."},
+		{v: []compose{{b: true, i: 123, s: "abc"}, {b: false, i: 456, s: "def"}}, n: 99, expect: "[{b:true i:123 s:abc} {b:false i:456 s:def}]"},
+	}
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply pkg/utils/string_test.go]: added in https://github.com/pingcap/dm/pull/257/commits/02aa7c3d02ce604be1d3f8ec03e59aefbe60d7c2.

File: pkg/baseconn/conn.go (hunk 1 above)
Comment (csuzhangxc) [reply pkg/baseconn/conn.go]: any better suggestion?

File: loader/db.go (hunk 2 above)
Comment (csuzhangxc) [reply loader/db.go]: done in https://github.com/pingcap/dm/pull/257/commits/02aa7c3d02ce604be1d3f8ec03e59aefbe60d7c2.

File: pkg/utils/common.go (hunk 3 above)
Comment (csuzhangxc) [reply pkg/utils/common.go]: done in https://github.com/pingcap/dm/pull/257/commits/02aa7c3d02ce604be1d3f8ec03e59aefbe60d7c2.

**Comment by csuzhangxc** [comment]
@amyangfei @WangXiangUSTC PTAL again

File: pkg/baseconn/conn.go (hunk 1 above)
Comment (WangXiangUSTC) [reply pkg/baseconn/conn.go]: ok, no better suggestion

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**